src/Parallel/GlobalCache.hpp

```diff
@@ -119,6 +119,23 @@ struct MockGlobalCache {
   explicit MockGlobalCache(CkMigrateMessage* /*msg*/) {}
   virtual ~MockGlobalCache() = default;
   virtual void pup(PUP::er& /*p*/) {}
+
+  static bool isIrreducible() { return true; }
+
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=noreturn"
+#elif defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-noreturn"
+#endif  // defined(__GNUC__) && !defined(__clang__)
+  template <typename T>
+  static void contribute(T&& /*unused*/) {
```

Comment:
Why not mark it as noreturn instead of silencing the warning?

Reply:
Charm++ compatibility. I don't want to worry about having anything different in the interface than what Charm++ does.